src/coreclr/scripts/superpmi.py

```diff
@@ -712,29 +742,32 @@ def __collect_mc_files__(self):
-            print_platform_specific_environment_vars(self.coreclr_args, "SuperPMIShimPath", self.jit_path)
-            print_platform_specific_environment_vars(self.coreclr_args, "COMPlus_AltJit", "*")
-            print_platform_specific_environment_vars(self.coreclr_args, "COMPlus_AltJitNgen", "*")
-            print_platform_specific_environment_vars(self.coreclr_args, "COMPlus_AltJitName", self.collection_shim_name)
-            print_platform_specific_environment_vars(self.coreclr_args, "COMPlus_EnableExtraSuperPmiQueries", "1")
-            print_platform_specific_environment_vars(self.coreclr_args, "COMPlus_TieredCompilation", "0")
+            logging.debug("Starting collection.")
+            logging.debug("")
+            print_platform_specific_environment_vars(logging.DEBUG, self.coreclr_args, "SuperPMIShimLogPath", self.temp_location)
+            print_platform_specific_environment_vars(logging.DEBUG, self.coreclr_args, "SuperPMIShimPath", self.jit_path)
+            print_platform_specific_environment_vars(logging.DEBUG, self.coreclr_args, "COMPlus_AltJit", "*")
+            print_platform_specific_environment_vars(logging.DEBUG, self.coreclr_args, "COMPlus_AltJitNgen", "*")
+            print_platform_specific_environment_vars(logging.DEBUG, self.coreclr_args, "COMPlus_AltJitName", self.collection_shim_name)
+            print_platform_specific_environment_vars(logging.DEBUG, self.coreclr_args, "COMPlus_EnableExtraSuperPmiQueries", "1")
+            print_platform_specific_environment_vars(logging.DEBUG, self.coreclr_args, "COMPlus_TieredCompilation", "0")
             if self.coreclr_args.use_zapdisable:
-                print_platform_specific_environment_vars(self.coreclr_args, "COMPlus_ZapDisable", "1")
-                print_platform_specific_environment_vars(self.coreclr_args, "COMPlus_ReadyToRun", "0")
-            print("")
+                print_platform_specific_environment_vars(logging.DEBUG, self.coreclr_args, "COMPlus_ZapDisable", "1")
+                print_platform_specific_environment_vars(logging.DEBUG, self.coreclr_args, "COMPlus_ReadyToRun", "0")
+            logging.debug("")
 
             if self.collection_command is not None:
-                print("%s %s" % (self.collection_command, " ".join(self.collection_args)))
+                logging.info("Collecting using command:")
+                logging.info("  %s %s" % (self.collection_command, " ".join(self.collection_args)))
 
                 assert isinstance(self.collection_command, str)
                 assert isinstance(self.collection_args, list)
 
                 self.command = [self.collection_command,] + self.collection_args
-                proc = subprocess.Popen(self.command, env=env_copy)
-                proc.communicate()
+                proc = subprocess.Popen(self.command, env=env_copy, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
can we just update `run_and_log()` to take `env` and use it here as well?

Reply:
Yes, probably. I'll look into doing that.